app/controllers/holds_controller.rb

```diff
@@ -122,4 +178,36 @@ def past_date
 
       redirect_to holds_path
     end
+
+    def hold_lookup(hold_key)
+      hold_record = symphony_client.get_hold_info hold_key, current_user.session_token
+      parsed_hold = JSON.parse hold_record.body
+      Hold.new parsed_hold
+    end
+
+    def item_lookup(barcode)
+      item_record = symphony_client.get_item_info barcode, current_user.session_token
+      parsed_item = JSON.parse item_record.body
+      Item.new parsed_item
+    end
+
+    def authorize_create!
+      return unless missing_args?
+
+      raise HoldCreateException, 'Error'
+    end
+
+    def deny_create
+      flash[:error] = if missing_args?
+                        t 'myaccount.hold.place_hold.missing_params'
+                      else
+                        t 'myaccount.hold.place_hold.select_volumes'
+                      end
+
+      redirect_to new_hold_path(catkey: params[:catkey])
+    end
+
+    def missing_args?
+      params['pickup_library'] === 'Not set' || params['pickup_by_date'].blank?
```

Comment:
What do you think about changing this method to a `case`? Right now it looks like "if there are missing args then tell the user they need to fill out location and pickup by day" else tell them to select a volume. But it seems like volume (i.e. barcode) is an argument right? I'm thinking (psuedo code) something like:

```
flash[:error] = case params
 when params['pickup_library'] === 'Not set'
   'error message'
 when params['pickup_by_date'].blank?
   'error message'
 when params['barcodes'].blank?
   'error message'
```

Reply:
I will give this a try, I originally wanted to go with similar approach but can not recall right now why I ended up checking barcodes and other separately 🤔

Reply:
probably because of the whole `barcode` versus `barcodes` complexity?

Reply:
made some changes for this and committed